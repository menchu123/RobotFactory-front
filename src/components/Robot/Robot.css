.robot__image-file {
  object-fit: cover;
  object-position: center;
  width: 100%;
}

.robot {
  font-size: smaller;
  border-radius: 20px;
  overflow: hidden;
}
