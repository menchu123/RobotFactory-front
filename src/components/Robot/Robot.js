import "./Robot.css";

const Robot = () => {
  return (
    <li className="robot card text-white bg-dark">
      <div className="robot__image card-img-top">
        <img
          className="robot__image-file"
          src="https://images.newscientist.com/wp-content/uploads/2021/04/21160754/21-april_pepper-the-robot.jpg"
          alt=""
          width="200"
          height="150"
        />
      </div>
      <div className="robot__info card-body">
        <h5 className="robot__name card-title">NOMBRE: ROGELIO</h5>
        <div class="robot__data">
          <table className="table text-white">
            <thead>
              <tr>
                <th scope="col">CARACTERÍSTICAS</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>VELOCIDAD</td>
                <td>10</td>
              </tr>
              <tr>
                <td>RESISTENCIA</td>
                <td>1</td>
              </tr>
              <tr>
                <td>AÑO DE CREACIÓN</td>
                <td>2001</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </li>
  );
};

export default Robot;
