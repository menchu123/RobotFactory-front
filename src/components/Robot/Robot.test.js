import { render, screen } from "@testing-library/react";
import Robot from "./Robot";
import ReactTestRenderer from "react-test-renderer";

describe("Given a Robot component", () => {
  describe("When it is passed a robot object with the name 'Rogelio'", () => {
    test("Then it should render a card with the robot's name", () => {
      const name = "Rogelio";
      const robot = {
        características: {
          velocidad: 0,
          resistencia: 10,
          creación: "2021",
        },
        _id: "61855440a99aeba4d99148ef",
        nombre: name,
        imagen: "https://cdn.kapwing.com/video_image-uBSRyuvqm.jpeg",
      };

      render(<Robot robot={robot} />);

      const robotName = screen.getByRole("heading");

      expect(robotName).toBeInTheDocument();
    });
  });
  describe("When it recives and object", () => {
    test("then it should render a card with the object info inside", () => {
      const robot = {
        características: {
          velocidad: 0,
          resistencia: 10,
          creación: "2021",
        },
        _id: "61855440a99aeba4d99148ef",
        nombre: "Rogelio",
        imagen: "https://cdn.kapwing.com/video_image-uBSRyuvqm.jpeg",
      };

      const robotCard = ReactTestRenderer.create(<Robot robot={robot} />);
      expect(robotCard.toJSON()).toMatchSnapshot();
    });
  });
});
