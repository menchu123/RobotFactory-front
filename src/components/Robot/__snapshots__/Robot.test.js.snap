// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Robot component When it recives and object then it should render a card with the object info inside 1`] = `
<div
  className="robot card text-white bg-dark col"
>
  <div
    className="robot__image card-img-top"
  >
    <img
      alt=""
      className="robot__image-file"
      height="250"
      src="https://cdn.kapwing.com/video_image-uBSRyuvqm.jpeg"
    />
  </div>
  <div
    className="robot__info card-body"
  >
    <h5
      className="robot__name card-title"
    >
      Rogelio
    </h5>
    <div
      className="robot__data"
    >
      <table
        className="table text-white"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              CARACTERÍSTICAS
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              VELOCIDAD
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              RESISTENCIA
            </td>
            <td>
              10
            </td>
          </tr>
          <tr>
            <td>
              AÑO DE CREACIÓN
            </td>
            <td>
              2021
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="buttons mb-3 mx-3"
  >
    <button
      className="btn btn-warning m-2"
    >
      Edit
    </button>
    <button
      className="btn btn-danger m-2"
    >
      Delete
    </button>
  </div>
</div>
`;
